containers/Dockerfile.ironic-nautobot-client

```diff
@@ -0,0 +1,32 @@
+ARG BASE=ghcr.io/rackerlabs/understack/argo-python3.11.8-alpine3.19:latest
+
+FROM ${BASE} AS builder
+ARG APP_PATH=/app
+ARG APP_USER=appuser
+ARG APP_GROUP=appgroup
+ARG APP_USER_UID=1000
+ARG APP_GROUP_GID=1000
+RUN --mount=type=cache,target=/var/cache/apk apk add --virtual build-deps gcc python3-dev musl-dev linux-headers
+RUN --mount=type=cache,target=/root/.cache/.pip pip install 'wheel==0.43.0'
+RUN --mount=type=cache,target=/root/.cache/.pip \
+    python -m venv /opt/poetry && \
+    /opt/poetry/bin/pip install 'poetry==1.7.1' && \
+    /opt/poetry/bin/poetry self add 'poetry-dynamic-versioning[plugin]==1.3.0'
+
+# copy in the code
+COPY --chown=${APP_USER}:${APP_GROUP} python/understack-workflows /app
+# need netifaces built as a wheel
+RUN --mount=type=cache,target=/root/.cache/.pip pip wheel --wheel-dir /app/dist netifaces
+# build wheels and requirements.txt, skip hashes due to building of netifaces above which won't match
+RUN cd /app && /opt/poetry/bin/poetry build -f wheel && /opt/poetry/bin/poetry export --without-hashes -f requirements.txt -o dist/requirements.txt
+
+FROM ${BASE} AS prod
+
+WORKDIR /app
+
+RUN mkdir -p /opt/venv/wheels/
+COPY --from=builder /app/dist/*.whl /app/dist/requirements.txt /opt/venv/wheels/
+
+USER $APP_USER
```

Comment:
this does not have any effect because the `$APP_USER` has been reset by the `FROM` on line 23:

```
❯ docker run --rm -it ghcr.io/rackerlabs/understack/ironic-nautobot-client:latest whoami
root
```